### PR TITLE
{math}[intel-2016a] Qhull 2015.2 (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

File: easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+homepage = 'http://www.qhull.org'
+description = """
+Qhull computes the convex hull, Delaunay triangulation, Voronoi diagram, halfspace intersection about a point,
+furthest-site Delaunay triangulation, and furthest-site Voronoi diagram. The source code runs in 2-d, 3-d, 4-d,
+and higher dimensions. Qhull implements the Quickhull algorithm for computing the convex hull.
+"""
+
+toolchain = {'name': 'intel', 'version': '2016a'}
+
+#recently checkged the Unix version numbering see https://github.com/Homebrew/homebrew-science/issues/3353
```
Comment (boegel) [inline easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb]: point to http://www.qhull.org/src/Changes.txt instead?

also, use space after `#` (our automatic style checking thing we're working on will trip over not doing so)

Comment (boegel) [inline easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb]: also, typo on `checkged`, should be `changed`?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2723.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/de9ba5e7bfd2f61e365f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2091.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/df990fb6284df346ab9e for a full test report.

**Comment by boegel** [comment]
@hajgato: looks good, just a minor style remark to look into

File: easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb (hunk 1 above)
Comment (hajgato) [inline easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb]: @boegel: Its not so clean for me, the windows version kept its download name (2015.2) and version (2015.2), the unix one changed its download name (2015-src-7.2.0), but it kept its base directory as `qhull-2015.2`. So what I think that version numbering is a mess, and they do not explain that in the `Changes.txt`.  
To be more precise, in the past they used `qhull_VERSION2` (2015.2), and internally `qhull_VERSION` (7.2.0) now it is a mixture  of the two in download names.

Honestly the whole issue does not worth this amount of writing, I am not even sure why the developers changed the download name.

File: easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Qhull computes the convex hull, Delaunay triangulation, Voronoi diagram, halfspace intersection about a point,
+furthest-site Delaunay triangulation, and furthest-site Voronoi diagram. The source code runs in 2-d, 3-d, 4-d,
+and higher dimensions. Qhull implements the Quickhull algorithm for computing the convex hull.
+"""
+
+toolchain = {'name': 'intel', 'version': '2016a'}
+
+# Recently changed the Unix version numbering see http://www.qhull.org/src/Changes.txt
+# and https://github.com/Homebrew/homebrew-science/issues/3353
+sources = ['%(namelower)s-2015-src-7.2.0.tgz']
```
Comment (boegel) [inline easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb]: Any idea if there's a relation to `2015.2` and `7.2.0`?

I'm just worried that we'll forget updating this is we bump the version...

Should we use this?

```python
sources = ['%(namelower)s-%(version_major)s-src-7.%(version_minor)s.0.tgz']
```

I'm making assumptions here, but at least we'll catch that we need to update this accordingly with a future version bump?

Comment (hajgato) [inline easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb]: it seems that:
2015.2 = 7.2.0
2015.1 = 7.1.0
2015.0.7 = 7.0.7
2015.0.6 = 7.0.6.2013
2015.0.5 = 7.0.5.1995
2015.0.4 = 7.0.4.1984
2015.0.3
2015.0.2
2015.0.1
2012.1  6.3.1.1494 (2012/02/18)
2012.1  6.3.1.1490 (2012/02/18)
2012.1  6.3.0.1483 (2012/02/02)
etc.

so maybe we have to refer as `qhull_VERSION`, and this is not consistent at all.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2668.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/94737d7ffce865793d84 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2090.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/231cef124438f66cb6d2 for a full test report.

File: easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Qhull computes the convex hull, Delaunay triangulation, Voronoi diagram, halfspace intersection about a point,
+furthest-site Delaunay triangulation, and furthest-site Voronoi diagram. The source code runs in 2-d, 3-d, 4-d,
+and higher dimensions. Qhull implements the Quickhull algorithm for computing the convex hull.
+"""
+
+toolchain = {'name': 'intel', 'version': '2016a'}
+
+# Recently changed the Unix version numbering see http://www.qhull.org/src/Changes.txt
+# and https://github.com/Homebrew/homebrew-science/issues/3353
+sources = ['%(namelower)s-%(version_major)-src-7.%(version_minor).0.tgz']
```
Comment (boegel) [inline easybuild/easyconfigs/q/Qhull/Qhull-2015.2-intel-2016a.eb]: you're missing the trailing `s` in `%(version_major)s` and `%(version_minor)s`

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2668.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/03aef7176c9be22157b8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2090.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/d86a2703e967a80cbded for a full test report.

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Pull request merged**